src/SFA.DAS.EmployerIncentives.Commands/UpsertLegalEntity/UpsertLegalEntityCommandHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Domain.Accounts;
+using SFA.DAS.HashingService;
+
+namespace SFA.DAS.EmployerIncentives.Commands.UpsertLegalEntity
+{
+    public class UpsertLegalEntityCommandHandler : ICommandHandler<UpsertLegalEntityCommand>
+    {
+        private readonly IAccountDomainRepository _domainRepository;
+        private readonly IHashingService _hashingService;
+
+        public UpsertLegalEntityCommandHandler(IAccountDomainRepository domainRepository, IHashingService hashingService)
+        {
+            _domainRepository = domainRepository;
+            _hashingService = hashingService;
+        }
+
+        public async Task Handle(UpsertLegalEntityCommand command, CancellationToken cancellationToken = default)
+        {
+            var account = await _domainRepository.Find(command.AccountId);
```

Comment:
Would be better if this published the AddCommand or updatedCommand depending on if the account exists.  The work don in EI-49 would make this easier to implement.